### PR TITLE
feat(datasources): add bitrise datasource

### PR DESCRIPTION
**Pull request by secustor**
## Changes
Adds a Bitrise datasource

which uses https://github.com/bitrise-io/bitrise-steplib as source similar to the BitriseCLI
~~Added a helper function to `getRawFile` to `GithubHttp` and zod schemas for API calls to the `/content` endpoint.~~
 

## Context
Allows implementation of https://github.com/renovatebot/renovate/issues/21957

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/http/github.ts [hunk 1]
```diff
@@ -493,4 +493,12 @@ export class GithubHttp extends Http<GithubHttpOptions> {
 
     return result;
   }
+
+  public getRawFile(url: string): Promise<HttpResponse> {
```
Comment (viceice) [inline lib/util/http/github.ts:497]: can we split this to a separate PR?

File: lib/modules/platform/github/schema.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { z } from 'zod';
```
Comment (viceice) [inline lib/modules/platform/github/schema.ts:1]: can we split this to a separate PR?

File: lib/modules/datasource/bitrise/index.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    const client = new GithubHttp(this.id);
```
Comment (viceice) [inline lib/modules/datasource/bitrise/index.ts:49]: do in constructor and override default http property

File: lib/modules/datasource/bitrise/index.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+    const parsedUrl = parseGitUrl(registryUrl);
+    if (detectPlatform(registryUrl) !== 'github') {
+      logger.warn(
+        `${parsedUrl.source} is not a supported Git hoster for this datasource`,
```
Comment (viceice) [inline lib/modules/datasource/bitrise/index.ts:54]: I think we should validate GitHub here, as the code below only works for GitHub not for ghes

File: lib/modules/datasource/bitrise/index.ts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // istanbul ignore if
+    if (!registryUrl) {
+      return null;
+    }
+
+    const parsedUrl = parseGitUrl(registryUrl);
+    if (detectPlatform(registryUrl) !== 'github') {
+      logger.warn(
```
Comment (rarkins) [inline lib/modules/datasource/bitrise/index.ts:56]:
```suggestion
      logger.once.warn(
```

File: lib/modules/datasource/bitrise/index.ts [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+    // if we have no releases return null
+    if (!result.releases.length) {
+      return null;
+    }
+
+    return {
+      ...result,
+      homepage: `https://bitrise.io/integrations/steps/${packageName}`,
+    };
```
Comment (rarkins) [inline lib/modules/datasource/bitrise/index.ts:136]:
```suggestion
    result.homepage = `https://bitrise.io/integrations/steps/${packageName}`;
    return result;
```

Isn't this more efficient, or does it need this shallow copy?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/datasource/bitrise/index.ts (hunk 6 above)
Comment (secustor) [reply lib/modules/datasource/bitrise/index.ts:136]: As the objects in question are minimal in size, the impact is nearly not existing. 
Doing that with RenovateConfig in a loop would be a different story.

**Review by rarkins — APPROVED** [review]
